org.eclipse.lsp4e/src/org/eclipse/lsp4e/enablement/EnablementTester.java

```diff
@@ -56,10 +56,13 @@ public String getDescription() {
 	 *
 	 * @return true if expression evaluates to true, false otherwise
 	 */
-	public boolean evaluate() {
+	public boolean evaluate(Object openResource) {
 		try {
 			final var context = new EvaluationContext(parent.get(), new Object());
 			context.setAllowPluginActivation(true);
+			if (openResource != null) {
+				context.addVariable("openResource", openResource); //$NON-NLS-1$
```

Comment:
LSP4E manipulates more than resources. The common item is the IDocument. Would it be possible to use it instead?

Reply:
The idea is, that the object behind the _openResource_ variable could be any object like `IFile`, `IDocument` or `IEditorInput`  and that the `enabledWhen` tester for `contentTypeMapping` in the  `org.eclipse.lsp4e.languageServer` extension point can  handle this. In my case a PropertyTester checks the object type and handles it appropriately (similar to HasLanguageServerPropertyTester.java)

Maybe the variable name could be better like _openObject_ or _object_ or _manipulatedObject_ ...

Reply:
> LSP4E manipulates more than resources. The common item is the IDocument. Would it be possible to use it instead?

Using IDocument now in evaluation contest as default variable now